data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,598 @@
+        required=True)
+    subparsers = parser.add_subparsers(dest='cmd')
+    setup_parser = subparsers.add_parser('setup')
+    setup_parser.add_argument(
+        'concept_lookup_dest_table',
+        action='store',
+        help='Table where suppressed concepts should be loaded')
+    retract_parser = subparsers.add_parser('retract')
+    retract_parser.add_argument(
+        '-s',
+        '--sandbox_dataset_id',
+        action='store',
+        dest='sandbox_dataset_id',
+        help='Identifies the dataset where output is stored',
+        required=True)
+
+    retract_parser.add_argument(
+        '-c',
+        '--concept_lookup_table',
+        action='store',
+        dest='concept_lookup_table',
+        help='Table where suppressed concepts are loaded',
+        required=True)
+
+    retract_parser.add_argument('-d',
+                                '--dataset_ids',
+                                action='store',
+                                dest='dataset_ids',
+                                nargs='+',
+                                help='Identifies dataset(s) to retract from',
+                                required=True)
+    return parser
+
+
+def setup(concept_lookup_table):
+    """
+    Load BQ tables needed before running
+
+    :param concept_lookup_table: destination table to save lookup
+    """
+    with open(DC732_CONCEPT_LOOKUP_CSV_PATH, 'rb') as csv_file:
+        job_config = bigquery.LoadJobConfig()
+        job_config.source_format = bigquery.SourceFormat.CSV
+        job_config.skip_leading_rows = 1
+        job_config.autodetect = True
+        job = CLIENT.load_table_from_file(csv_file,
+                                          destination=concept_lookup_table,
+                                          job_config=job_config)
+        job.result()
+        print("Loaded {} rows into {}.".format(job.output_rows,
+                                               concept_lookup_table))
+
+
+INVALID_TABLE_REF = (
+    "Invalid table reference: '{table}' "
+    "(qualify with either '<dataset>.<table>' or '<project>.<dataset>.<table>')"
+)
+
+
+def validate_lookup_table_arg(table):
+    """
+    Raise an error if the table reference is not qualified by dataset or project and dataset
+
+    :param table: reference to a table
+    """
+    dot_count = table.count('.')
+    if dot_count < 1 or dot_count > 2:
+        raise ValueError(INVALID_TABLE_REF.format(table=table))
+
+
+def parse_args(args):
+    """
+    Parse command line arguments
+
+    :return: namespace with parsed arguments
+    """
+    parser = get_arg_parser()
+    args = parser.parse_args(args)
+    if args.cmd == 'setup':
+        validate_lookup_table_arg(args.concept_lookup_dest_table)
+    elif args.cmd == 'retract':
+        validate_lookup_table_arg(args.concept_lookup_table)
+    return args
+
+
+if __name__ == '__main__':
+    import sys
+
+    ARGS = parse_args(sys.argv[1:])
```

Comment:
why the restriction here?  what are you cutting out of the command line arguments?

Reply:
`sys.argv[0]` is the program name. I wanted the signature for `parse_args()` to be the same as `ArgumentParser.parse_args()` which it wraps. If we don't strip the program name here then we'll have to do it in the implementation. Requiring 'dc732_suppress.py' as the first element would make the interface less intuitive to me, but I would feel ok with it if we name the function `__parse_args()`. Let me know what you think.

Reply:
I think you can achieve the same thing by:

```
def parse_args(args=None):
   # Then your function code here.  If you pass a list of command line arguments, that is
    # what gets parsed.  If you pass None, the command line arguments are read by default.
   # this pattern is used by deid/parser.py parse_args to either read from the command line or
   # accept a parameter list from tools/run_deid.py

# Using None tells ArgumentParser.parse_args() to read from the command line
ARGS = parse_args()
```

Reply:
Cool, I will check that out